lib/util.js

```diff
@@ -1022,24 +1027,27 @@ class Load {
    * @return {Object} - The module level configuration object.
    */
   setModuleDefaults(moduleName, defaultProperties) {
-    const moduleConfig = Util.cloneDeep(defaultProperties);
-    let moduleDefaults;
+    if (this.defaults === undefined) {
+      this.defaults = {};
+      this.unmerged = {};
 
-    if (this.sources === undefined) {
-      moduleDefaults = {};
-    } else if (this.sources.length > 0 && this.sources[0].name === 'Module Defaults') {
-      moduleDefaults = this.sources[0].parsed;
-    } else {
-      let source = { name: 'Module Defaults', parsed: {} };
-      this.sources.splice(0, 0, source);
-      moduleDefaults = source.parsed;
+      if (this.sources) {
+        this.sources.splice(0, 0, { name: 'Module Defaults', parsed: this.defaults });
+      }
     }
 
     const path = moduleName.split('.');
-    Util.setPath(moduleDefaults, path, Util.cloneDeep(defaultProperties)); // TODO: This clobber is from the original code, and is a bug
+    const defaults = Util.setPath(this.defaults, path, Util.getPath(this.defaults, path) ?? {});
+
+    Util.extendDeep(defaults, defaultProperties);
 
-    Util.extendDeep(moduleConfig, Util.getPath(this.config, path) || {});
+    const original =
```

Comment:
Snapshot the config data as it existed before we added defaults to it, so that duplicate calls to set defaults can be resolved in chronological order.